wallet/chain/c/builder_with_options.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package c
+
+import (
+	"math/big"
+
+	"github.com/ava-labs/coreth/plugin/evm"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+	"github.com/ava-labs/avalanchego/wallet/subnet/primary/common"
+)
+
+var _ Builder = (*builderWithOptions)(nil)
+
+type builderWithOptions struct {
+	Builder
+	options []common.Option
+}
+
+// NewBuilderWithOptions returns a new transaction builder that will use the
+// given options by default.
+//
+//   - [builder] is the builder that will be called to perform the underlying
+//     operations.
+//   - [options] will be provided to the builder in addition to the options
+//     provided in the method calls.
+func NewBuilderWithOptions(builder Builder, options ...common.Option) Builder {
```

Comment:
> Why is a a builder with options suggested

I wouldn't say it is suggested... just that it is useful. Example: we write 2 scripts that issue a bunch of txs from a wallet. We want to execute these scripts in parallel using a single wallet... You can wrap the wallet with options so that the prior scripts do not need to know about each other. (The builder with options is then used just to correctly implement the `Builder()` function on the wallet interface.